### PR TITLE
update SMTP auth config attribute for keycloak_realm resource

### PR DESCRIPTION
**Pull request by mrparkers**
cc @tomrutsaert

### CODE REVIEW

File: keycloak/realm.go [hunk 1]
```diff
@@ -46,18 +46,18 @@ type Realm struct {
...
-	Host               string `json:"host,omitempty"`
-	ReplyTo            string `json:"replyTo,omitempty"`
-	ReplyToDisplayName string `json:"replyToDisplayName,omitempty"`
-	From               string `json:"from,omitempty"`
-	FromDisplayName    string `json:"fromDisplayName,omitempty"`
-	EnvelopeFrom       string `json:"envelopeFrom,omitempty"`
-	Ssl                string `json:"ssl,omitempty"`
-	User               string `json:"user,omitempty"`
-	Password           string `json:"password,omitempty"`
+	StartTls           KeycloakBoolQuoted `json:"starttls,omitempty"`
```
Comment (mrparkers) [inline keycloak/realm.go]: This is a special type that serializes to a JSON string of `"true"` or `"false"`, which is helpful for parts of the Keycloak API that expect boolean strings.  Under the hood it's a regular boolean that is easier to work with in the provider code.

File: provider/resource_keycloak_realm.go [hunk 2]
```diff
@@ -76,17 +76,13 @@ func resourceKeycloakRealm() *schema.Resource {
 			//Smtp server
 
 			"smtp_server": {
-				Type:     schema.TypeSet,
+				Type:     schema.TypeList,
```
Comment (mrparkers) [inline provider/resource_keycloak_realm.go]: This won't change how this attribute is used, but it's much easier to use `TypeList` for one-off block attributes.

File: provider/resource_keycloak_realm.go [hunk 3]
```diff
@@ -425,9 +452,17 @@ func setRealmData(data *schema.ResourceData, realm *keycloak.Realm) {
...
+		if realm.SmtpServer.Auth {
+			auth := make(map[string]interface{})
+
+			auth["username"] = realm.SmtpServer.User
+			auth["password"] = realm.SmtpServer.Password
+
+			smtpSettings["auth"] = []interface{}{auth}
+		}
+
+		data.Set("smtp_server", []interface{}{smtpSettings})
```
Comment (mrparkers) [inline provider/resource_keycloak_realm.go]: This line is why `TypeList` is easier to work with than `TypeString` in my opinion.

File: keycloak/realm.go (hunk 1 above)
Comment (tomrutsaert) [reply keycloak/realm.go]: Ok, good to know, I will use that in new development

File: provider/resource_keycloak_realm.go (hunk 2 above)
Comment (tomrutsaert) [reply provider/resource_keycloak_realm.go]: ok

File: provider/resource_keycloak_realm.go (hunk 3 above)
Comment (tomrutsaert) [reply provider/resource_keycloak_realm.go]: I will be using this.

File: docs/resources/keycloak_realm.md [hunk 4]
```diff
@@ -71,6 +81,24 @@ The attributes below should be specified as [Go duration strings](https://golang
...
 - `action_token_generated_by_admin_lifespan` - (Optional) The maximum time a user has to use an admin-generated permit before it expires.
 
+##### SMTP
+
+The `smtp_server` block can be used to configure the realm's SMTP settings, which can be found in the "Email" tab in the GUI.
+This block supports the following attributes:
+
+- `host` - (Required) The host of the SMTP server.
+- `port` - (Optional) The port of the SMTP server (defaults to 25).
+- `from` - (Optional) The email address for the sender.
```
Comment (tomrutsaert) [inline docs/resources/keycloak_realm.md]: Isn't "from" a required field?

Comment (mrparkers) [reply docs/resources/keycloak_realm.md]: Good catch!

**Pull request merged**